lib/scala/runtime-version-manager/src/main/scala/org/enso/runtimeversionmanager/runner/NativeJavaCommand.scala

```diff
@@ -0,0 +1,20 @@
+package org.enso.runtimeversionmanager.runner
+
+import org.enso.cli.OS
+import org.enso.distribution.{DistributionManager, Environment}
+
+import java.nio.file.Path
+
+case class NativeJavaCommand(executablePath: Path)
+    extends JavaCommand(executablePath.toString, None)
+
+object NativeJavaCommand {
+  def apply(version: String): NativeJavaCommand = {
+    val env      = new Environment() {}
+    val dm       = new DistributionManager(env)
+    val execName = OS.executableName("enso")
+    val fullExecPath =
+      dm.paths.engines.resolve(version).resolve("bin").resolve(execName)
```

Comment:
I'd remove all the support of `--native-image` options and just checked for existence of this `fullExecPath` binary (probably check it is not a bash shell) and if present just invoke it. It should be good enough for development purposes, but please note that it is also correct for the future:
- the [discussion entry](https://github.com/orgs/enso-org/discussions/10121#discussioncomment-10272101) outlines we should _replace launchers with binary ones_
- when `--jvm` option is specified the `bin/enso` or `bin\enso.exe` should switch to JVM mode themselves - already implemented since #11500 

E.g. preferring the `enso` binary if it exists is the right solution for the future.